buildkite/src/Jobs/Lint/OCaml/Spec.dhall

```diff
@@ -1,8 +1,12 @@
+let S = ../../../Lib/SelectFiles.dhall
 let JobSpec = ../../../Pipeline/JobSpec.dhall
 
 in
 JobSpec::{
-  dirtyWhen = "^buildkite/src/Jobs/Lint/OCaml|^src/",
+  dirtyWhen = [
+    S.contains "^buildkite/src/Jobs/Lint/OCaml",
+    S.contains "^src/"
```

Comment:
Shouldn't we abstract over "^" and "$"? For instance, `S.starts_with "src/" ==> "^src/.*$"`, `S.contains "src/" ==> "^.*/src/.*$"`, and you can add other variants for different cases. If you go this direction, I would recommend representing the "^" and "$" entirely in `S.Type` (for instance, `S::{ dir=_, exts=_, opt=StrictStart }`, or something like `S::{ dir=_, exts=_, strict_start=True, strict_end=True }`). Then you would just escape any "^" or "$" in the dir strings to protect against ill formatted output.

Reply:
Dhall can't escape characters for you afaik (rule of thumb is once you have a text it's opaque). So no way to escape those chars. If we think of how we'll use it: I don't think we'll need to support files with special chars in them.

Good call -- I already stick a $ at the end of a file when you have an extension. I will make that change with the option.

Reply:
Oh, I see. Didn't realize that `Text` cannot be inspected in Dhall (for instance, `Text/toList`).